### PR TITLE
[#3764] improvement(docs): Add user docs for using GVFS in Python

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Provides documentation for users to use Gravitino Virtual FileSystem in Python.

### Why are the changes needed?

Fix: #3764 

### How was this patch tested?

No code changes, no testing required.

### CODE REVIEW

**Comment by xloya** [comment]
@jerryshao Could you take a look of this? I wrote this user doc with reference to Hadoop GVFS. Thanks.

File: docs/how-to-use-python-gvfs.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
+---
+title: How to use Gravitino Virtual File System with Filesets in Python
+slug: /how-to-use-python-gvfs
+license: "Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
```
Comment (jerryshao) [inline docs/how-to-use-python-gvfs.md:5]: The license header should be updated to use ASF one, please refer to other docs.

File: docs/how-to-use-python-gvfs.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+---
+
+## Introduction
```
Comment (jerryshao) [inline docs/how-to-use-python-gvfs.md:8]: Maybe we don't need to create another doc for python gvfs, add into the current `hot-to-use-gvfs.md` should be OK, WDYT?

Comment (xloya) [reply docs/how-to-use-python-gvfs.md:8]: It' OK.

Comment (xloya) [reply docs/how-to-use-python-gvfs.md]: Added the new content to the original GVFS doc, please take a look, thanks.

File: docs/how-to-use-gvfs.md [hunk 3]
```diff
@@ -21,6 +23,10 @@ the files or folders as shown below:
 ```text
 gvfs://fileset/${catalog_name}/${schema_name}/${fileset_name}/sub_dir/
 ```
+In python GVFS, you can also access the files or folders as shown below:
```
[the remark this answers is not in the document]
Comment (xloya) [reply docs/how-to-use-gvfs.md:26]: I see.

File: docs/how-to-use-gvfs.md [hunk 4]
```diff
@@ -307,3 +315,210 @@ conf.set("fs.gravitino.client.kerberos.keytabFilePath", "${your_kerberos_keytab}
...
+```
+
+### Usage examples
+
+1. Make sure to obtain the Gravitino library.
+   You can get it by [pip](https://pip.pypa.io/en/stable/installation/):
+
+```shell
+pip install gravitino
+```
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:359]: Can you please indent this code block to align with the paragraph above.

File: docs/how-to-use-gvfs.md [hunk 5]
```diff
@@ -307,3 +315,210 @@ conf.set("fs.gravitino.client.kerberos.keytabFilePath", "${your_kerberos_keytab}
...
+
+2. Configuring the Hadoop environment.
+   You should ensure that the Python client has Kerberos authentication information and
+   configure Hadoop environments in the system environment:
+```shell
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:364]: Add a blank line above and make this code block indentation.

File: docs/how-to-use-gvfs.md [hunk 6]
```diff
@@ -307,3 +315,210 @@ conf.set("fs.gravitino.client.kerberos.keytabFilePath", "${your_kerberos_keytab}
...
+
+fs = gvfs.GravitinoVirtualFileSystem(
+    server_uri="http://localhost:8090", metalake_name="test_metalake"
+)
+
+# read a parquet file as ray dataset
+ds = ray.data.read_parquet("gvfs://fileset/fileset_catalog/tmp/tmp_fileset/test.parquet",fs)
+```
+
+4. Integrating with [Llama Index](https://docs.llamaindex.ai/en/stable/module_guides/loading/simpledirectoryreader/#support-for-external-filesystems)(0.10.40).
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:509]: "LlamaIndex", not "Llama Index".

File: docs/how-to-use-gvfs.md [hunk 7]
```diff
@@ -307,3 +315,210 @@ conf.set("fs.gravitino.client.kerberos.keytabFilePath", "${your_kerberos_keytab}
...
+```python
+from gravitino import gvfs
+import pandas as pd
+
+fs = gvfs.GravitinoVirtualFileSystem(
+    server_uri="http://localhost:8090", metalake_name="test_metalake"
+)
+
+# read a parquet file as arrow dataset
+arrow_dataset = dt.dataset("gvfs://fileset/fileset_catalog/tmp/tmp_fileset/test.parquet", filesystem=fs)
```
Comment (shaofengshi) [inline docs/how-to-use-gvfs.md:490]: what's the "dt" here, shall it be "pd"?

Comment (xloya) [reply docs/how-to-use-gvfs.md:496]: There is a problem with the code here, update it to PyArrow lib.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**